brokenspoke_analyzer/core/utils.py

```diff
@@ -48,14 +49,42 @@ def gunzip(
     # Decompress it.
     gzip_file = gzip_file.resolve(strict=True)
     with gzip.open(gzip_file, "rb") as f:
-        content = f.read()
-        target.resolve().write_bytes(content)
+        try:
+            content = f.read()
+        except gzip.BadGzipFile:
+            logger.debug("BadGzipFile")
+            if delete_after:
+                gzip_file.unlink()
+        except EOFError:
+            logger.debug("EOFError")
+            if delete_after:
+                gzip_file.unlink()
+        else:
+            target.resolve().write_bytes(content)
 
     # Delete the archive.
     if delete_after:
         gzip_file.unlink()
 
 
+def file_checksum_ok(osm_file: pathlib.Path, osm_file_md5: pathlib.Path) -> bool:
+    BUF_SIZE = 65536
+    HASH_SIZE = 32  # 128 bit MD5 hash
+    md5 = hashlib.md5()
+
+    with open(osm_file, "rb") as f:
+        while True:
+            data = f.read(BUF_SIZE)
+            if not data:
+                break
+            md5.update(data)
+
+    with open(osm_file_md5, "r") as f:
+        readchecksum = f.read(HASH_SIZE)
```

Comment:
Same thing here, we could leverage the `pathlib.Path` API:

```suggestion
    readchecksum = osm_file_md5.read_text()
```

See https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text

Reply:
:+1: